src/clm/commands/inner_tabulate_molecules.py

```diff
@@ -60,20 +62,22 @@ def tabulate_molecules(input_file, train_file, representation, output_file):
             mass = round(Descriptors.ExactMolWt(mol), 6)
             formula = rdMolDescriptors.CalcMolFormula(mol)
             canonical_smile = Chem.MolToSmiles(mol, isomericSmiles=False)
-
-            if canonical_smile not in train_smiles:
-                new_smiles.append([canonical_smile, mass, formula])
-
-    freqs = (
-        pd.DataFrame(new_smiles, columns=["smiles", "mass", "formula"])
-        .groupby(["smiles", "mass", "formula"])
-        .size()
-        .to_frame("size")
-        .sort_values("size", ascending=False)
-        .reset_index()
+            inchikey = Chem.inchi.MolToInchiKey(mol)
+
+            if inchikey not in train_data:
+                new_smiles.append([canonical_smile, mass, formula, inchikey])
+
+    freqs = pd.DataFrame(new_smiles, columns=["smiles", "mass", "formula", "inchikey"])
```

Comment:
slightly more verbose than before because we wish to group by inchi/mass/formula, and also have the canonical smile (first, as per this implementation) be present in the dataframe after grouping.